crates/biome_js_analyze/src/lint/nursery/use_google_font_display.rs

```diff
@@ -0,0 +1,137 @@
+    ///
+    /// This rule flags the absence of the `font-display` parameter, or the usage of less optimal values such as `auto`, `block`, or `fallback`.
+    /// Using `&display=optional` is generally recommended as it minimizes the risk of invisible text or layout shifts.
+    /// In cases where swapping to the custom font after it has loaded is important, consider using `&display=swap`.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=auto" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=block" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=fallback" />
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=optional" rel="stylesheet" />
+    /// ```
+    ///
+    /// ```jsx
+    /// <link href="https://fonts.googleapis.com/css2?display=unknown" rel="stylesheet" />
+    /// ```
+    ///
+    /// ```jsx
+    /// <link rel="stylesheet" />
+    /// ```
+    pub UseGoogleFontDisplay {
+        version: "next",
+        name: "useGoogleFontDisplay",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("google-font-display")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+    }
+}
+
+impl Rule for UseGoogleFontDisplay {
+    type Query = Ast<AnyJsxElement>;
+    type State = (bool, TextRange);
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let element = ctx.query();
+
+        if element.name().ok()?.name_value_token()?.text_trimmed() != "link" {
+            return None;
+        }
+
+        let href = element.find_attribute_by_name("href");
+        let initializer = href?.initializer()?.value().ok()?;
+        let jsx_string = initializer.as_jsx_string();
+        let href_text = jsx_string?.inner_string_text().ok()?;
+
+        if !href_text.starts_with("https://fonts.googleapis.com/css") {
+            return None;
+        }
+
+        let display_param = href_text
+            .text()
+            .split('?')
+            .last()?
+            .split('&')
+            .find(|p| p.starts_with("display="));
+        let range = jsx_string?.value_token().ok()?.text_trimmed_range();
+
+        if let Some(display_param) = display_param {
+            if matches!(
+                &*display_param.replace("display=", ""),
+                "auto" | "block" | "fallback"
+            ) {
```

Comment:
Instead of using a `.replace`, which requires a dereference and a manipulation, you could use a more static check:

```rs
const FORBIDDEN_VALUES: [&str; 3] = ["auto", "block", "fallback"];

fn run() {
	for forbidden_value in FORBIDDEN_VALUES {
		if display_param.ends_with(forbidden_value) {
			return Some()
		}
	}
}
```